ethcore/src/ethereum/ethash.rs

```diff
@@ -219,7 +252,7 @@ impl Engine<EthereumMachine> for Arc<Ethash> {
 			self.machine.add_balance(block, a, reward)?;
 		}
 
-		// note and trace.
+		// Note and trace.
 		self.machine.note_rewards(block, &[(author, result_block_reward)], &uncle_rewards)
```

Comment:
without altering this line, block rewards from this change won't be traced. I would recommend using a 
`Cow<&'a [(Address, U256)]>` to store the main block reward initially to avoid an allocation on every block unless it's a musicoin chain.

Reply:
or a separate call to `machine.note_rewards` within the musicoin if block would serve the same purpose.

Reply:
Should I call `machine.note_rewards` for _each_ block reward or only for the miner reward?

Reply:
`note_rewards` takes two slices of `&[(Address, U256)]`. The first one is for block rewards and the second for uncle rewards. I don't think it matters if you make multiple calls to the function.

@grbIzl tracing does handle multiple block rewards correctly, right?

Reply:
@rphmeier it should. It's pretty straightforward.

Reply:
good. @5chdn then you would say the correct implementation traces only the miner reward and not any of the others? LGTM if so.